Improve handling of fetch errors during login

kuosandys opened the pull request:
Towards https://github.com/giantswarm/giantswarm/issues/20464.

This PR improves the handling of fetch errors thrown while a user is logging into the application. While we can't reliably check for specific network errors (e.g. `net::ERR_CONNECTION_CLOSED`), these errors fall under the `TypeError` exception thrown by `fetch()`. We improve handling by including the original error message in the error reported to Sentry, and displaying a more appropriate error message to the user in this case during login:

<img width="337" alt="Screen Shot 2022-01-25 at 09 51 27" src="https://user-images.githubusercontent.com/62935115/150943641-251c7737-8f2d-4352-b778-410888b41eea.png">

One thing to bear in mind is that these errors can occur for a number of different reasons (more information can be found [here](https://developer.mozilla.org/en-US/docs/Web/API/fetch#exceptions)), so it wouldn't be appropriate to display a message that is too specific, and should rather prompt the user to reach out to support if this occurs during the login process.

marians commented:
Nice! I'd like to have something like "if this problem persists" in the message, to not trigger support queries for one-off issues.

kuosandys commented:
> Nice! I'd like to have something like "if this problem persists" in the message, to not trigger support queries for one-off issues.

@marians I've updated the error message to: "We couldn't execute a request. Please try again later, and contact support at support@giantswarm.io if the problem persists" (the screenshot above is also updated). What do you think?

marians reviewed (approved):
Great! Thank you
